File: dotnet-desktop-guide/winforms/controls/snippets/how-to-add-an-event-handler/cs/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace cs
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }

        private void label1_MouseEnter(object sender, EventArgs e)
        {
            label1.Text = "Mouse entered!";
        }

        private void label1_DoubleClick(object sender, EventArgs e)
        {
            label1.Text = "Doubleclicked!";
        }

        private void label1_PreviewKeyDown(object sender, PreviewKeyDownEventArgs e)
        {

        }

        private void textBox1_PreviewKeyDown(object sender, PreviewKeyDownEventArgs e)
        {
            label1.Text = "preview key";
            
        }

        private void button1_KeyPress(object sender, KeyPressEventArgs e)
        {
            label1.Text = "button key";

        }

        private void a(object sender, KeyPressEventArgs e)
        {
            label1.Text = "button key";

        }

        private void button1_KeyDown(object sender, KeyEventArgs e)
        {

        }

        private void button1_ChangeUICues(object sender, UICuesEventArgs e)
        {

        }

        //<HandlerViaCode>
        private void button1_Click(object sender, EventArgs e)
        {
            // Create and add the button
            Button myNewButton = new()
            {
                Location = new Point(10, 10),
                Size = new Size(120, 25),
                Text = "Do work"
            };

            // Handle the Click event for the new button
            myNewButton.Click += MyNewButton_Click;
            this.Controls.Add(myNewButton);

            // Remove this button handler so the user cannot do this twice
            //<RemoveHandler>
            button1.Click -= button1_Click;
            //</RemoveHandler>
        }

        private void MyNewButton_Click(object sender, EventArgs e)
        {
            
        }
        //</HandlerViaCode>
    }
}

File: dotnet-desktop-guide/winforms/controls/snippets/how-to-add-an-event-handler/vb/Form1.vb
Imports System.Windows.Forms
Imports System.Drawing

Partial Public Class Form1

    Public Sub New()
        InitializeComponent()
    End Sub

    Private Sub Button1_Click_2(sender As Object, e As EventArgs)
        '<CreateControl>
        Dim label1 As New Label With {.Text = "&First Name",
                                      .Location = New Point(10, 10),
                                      .TabIndex = 10}

        Dim field1 As New TextBox With {.Location = New Point(label1.Location.X,
                                                              label1.Bounds.Bottom + Padding.Top),
                                        .TabIndex = 11}

        Controls.Add(label1)
        Controls.Add(field1)
        '</CreateControl>
    End Sub

    '<HandlerViaCode>
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        'Create and add the button
        Dim myNewButton As New Button() With {.Location = New Point(10, 10),
                                              .Size = New Size(120, 25),
                                              .Text = "Do work"}

        'Handle the Click event for the new button
        AddHandler myNewButton.Click, AddressOf MyNewButton_Click
        Me.Controls.Add(myNewButton)

        'Remove this button handler so the user cannot do this twice
        '<RemoveHandler>
        RemoveHandler Button1.Click, AddressOf Button1_Click
        '</RemoveHandler>
    End Sub

    Private Sub MyNewButton_Click(sender As Object, e As EventArgs)

    End Sub
    '</HandlerViaCode>
End Class
